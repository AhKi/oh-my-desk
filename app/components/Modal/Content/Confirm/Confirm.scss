.Confirm {
  width: 300px;
  padding: 25px;
  overflow: auto;
  text-align: center;
}

