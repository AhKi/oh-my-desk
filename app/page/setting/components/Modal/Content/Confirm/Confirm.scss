.Confirm {
  width: 300px;
  padding: 25px;
  overflow: auto;
  text-align: center;
}

.Confirm__title {
  text-align: center;
  font-weight: 600;
  color: $color-middle-gray-blue;
}

.Confirm__content {
  font-weight: 400;
  font-size: 13px;
  color: $color-middle-gray-blue;
  display: block;
  text-align: center;
}

.Confirm__btn-box {
  display: flex;
  justify-content: center;
}
