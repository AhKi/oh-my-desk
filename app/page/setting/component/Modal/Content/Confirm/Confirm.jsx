import React from 'react';
import PropTypes from 'prop-types';
import './Confirm.scss';

const propTypes = {
  btnText: PropTypes.string,
  content: PropTypes.string,
  title: PropTypes.string,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  btnText: 'OK',
  content: '',
  title: '',
  onModalClose() {},
};

class Confirm extends React.Component {
  constructor(props) {
    super(props);
    this.handleEnterEvent = this.handleEnterEvent.bind(this);
  }

  componentDidMount() {
    document.addEventListener('keydown', this.handleEnterEvent);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleEnterEvent);
  }

  handleEnterEvent(e) {
    if (e.keyCode === 13) {
      this.props.onModalClose();
    }
  }

  render() {
    const {
      btnText,
      content,
      title,
      onModalClose,
    } = this.props;

    return (
      <div className="Confirm">
        <h3 className="Confirm__title">{title}</h3>
        <p className="Confirm__content">{content}</p>
        <button
          type="button"
          className="Btn Btn--primary Btn-middle"
          onClick={onModalClose}
        >
          {btnText}
        </button>
      </div>
    );
  }
}

Confirm.propTypes = propTypes;
Confirm.defaultProps = defaultProps;

export default Confirm;
