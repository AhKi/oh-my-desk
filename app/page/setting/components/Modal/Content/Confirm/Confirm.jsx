import React from 'react';
import PropTypes from 'prop-types';
import './Confirm.scss';

const propTypes = {
  btnText: PropTypes.string,
  content: PropTypes.string,
  title: PropTypes.string,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  btnText: 'OK',
  content: '',
  title: '',
  onModalClose() {},
};

class Confirm extends React.Component {
  constructor(props) {
    super(props);
    this.handleEnterEvent = this.handleEnterEvent.bind(this);
  }

  componentDidMount() {
    document.addEventListener('keydown', this.handleEnterEvent);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleEnterEvent);
  }

  handleEnterEvent(e) {
    const { onModalClose } = this.props;
    if (e.keyCode === 13) {
      onModalClose();
    }
  }

  render() {
    const {
      btnText,
      content,
      title,
      onModalClose,
    } = this.props;

    return (
      <div className="Confirm">
        <h6 className="Confirm__title space-2x">
          <strong>
            {title}
          </strong>
        </h6>
        <span
          className="Confirm__content space-4x"
        >
          {content.split('\n').map((line, index) => ( // eslint-disable-next-line react/no-array-index-key
            <span key={index}>
              {line}
              <br />
            </span>
          ))}
        </span>
        <div className="Confirm__btn-box">
          <button
            type="button"
            className="Btn Btn--primary Btn--sm"
            onClick={onModalClose}
          >
            {btnText}
          </button>
        </div>
      </div>
    );
  }
}

Confirm.propTypes = propTypes;
Confirm.defaultProps = defaultProps;

export default Confirm;
