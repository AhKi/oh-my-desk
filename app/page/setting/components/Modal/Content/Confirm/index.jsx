import { connect } from 'react-redux';
import { modalClose } from 'setting/store/modal/actions';

import Confirm from './Confirm';

const mapDispatchToProps = {
  onModalClose: modalClose,
};

export default connect(null, mapDispatchToProps)(Confirm);
