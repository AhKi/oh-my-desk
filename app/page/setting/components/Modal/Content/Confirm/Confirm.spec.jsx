import React from 'react';
import { shallow } from 'enzyme';

import Confirm from './Confirm';

describe('<Confirm />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Confirm />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('test handleEnterEvent', () => {
    const onModalClose = jest.fn();
    const wrapper = shallow(<Confirm onModalClose={onModalClose} />);

    afterEach(() => {
      onModalClose.mockClear();
    });

    it('when e.keyCode === 13', () => {
      wrapper.instance().handleEnterEvent({
        keyCode: 13,
      });

      expect(onModalClose).toHaveBeenCalledTimes(1);
      expect(onModalClose).toHaveBeenCalledWith();
    });

    it('when e.keyCode !== 13', () => {
      wrapper.instance().handleEnterEvent({
        keyCode: 14,
      });

      expect(onModalClose).toHaveBeenCalledTimes(0);
    });
  });
});
