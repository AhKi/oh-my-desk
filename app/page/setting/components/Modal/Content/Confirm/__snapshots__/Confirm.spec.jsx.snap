// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirm /> should match to snapshot when render default 1`] = `
<div
  className="Confirm"
>
  <h6
    className="Confirm__title space-2x"
  >
    <strong />
  </h6>
  <span
    className="Confirm__content space-4x"
  />
  <div
    className="Confirm__btn-box"
  >
    <button
      className="Btn Btn--primary Btn--sm"
      onClick={[Function]}
      type="button"
    >
      OK
    </button>
  </div>
</div>
`;
